Core/Core/CourseSync/Common/Models/Entities/CourseSyncEntry.swift

```diff
@@ -182,6 +182,14 @@ public struct CourseSyncEntry: Equatable {
             files.indices.forEach { files[$0].selectionState = selectionState }
         }
 
+        if selectionState == .selected {
+            if tabs[id: id]?.type == .modules || tabs[id: id]?.type == .grades {
+                for (index, tab) in tabs.enumerated() where tab.type != .modules && tab.type != .grades {
+                    tabs[index].selectionState = .selected
+                }
+            }
+        }
+
```

Comment:
Could we add a logic that de-selects grades and modules if any of the dependent items are de-selected? Now if I select modules and after de-select manually every other item except modules then it won't work as expected.

Reply:
Good idea, added.